### DISCUSSION TITLE
Failing docker builds lately

### DISCUSSION THREAD

**Discussion by madibaT** [ant-design/ant-design#42550] [General]
hey All,

We've upgraded to Ant 5.1.7 3 months ago. Our builds have been working fine but we're seeing this issue since last week.

is there anyone experiencing the same? i wanted to check before opening an issue.

TS2741: Property 'rev' is missing in type '{ style: { marginRight: number; }; }' but required in type 'Pick<AntdIconProps, "default" | "results" | "id" | "label" | "disabled" | "name" | "type" | "max" | "min" | "value" | "content" | "title" | "target" | "children" | ... 349 more ... | "twoToneColor">'.
#25 261.5       40 |         ) : (
#25 261.5       41 |           <>
#25 261.5     > 42 |             <BarsOutlined style={{ marginRight: 2 }} />
#25 261.5          |              ^^^^^^^^^^^^
#25 261.5       43 |             <DownOutlined />
#25 261.5       44 |           </>
#25 261.5       45 |         )}

**Comment by AshleyLTW** [comment]
I'm having the same issue here, I'm on 5.2.0 and using Netlify. 

The issue started when I tried to upgrade to 5.5.1. I downgraded to 5.2.0 and `npm run build` is working again, but it still fails when I push to netlify.

**Reply by madibaT** [reply]
We finally found a root cause.

it turns out that 'ant' package has a child dependency of 'ant-design' and it's referenced as "@ant-design/icons": "^5.0.0". This usually means 'ant-design' can install any 5.x version and it usually picks the latest.

@ant-design released a package released version 5.1.0 last week and this has the breaking change.

to fix this, we had to overwrite the dev dependencies like this in the package.json file:

"overrides": {
    "@ant-design/icons": "5.0.1"
  },

this worked.

**Reply by AshleyLTW** [reply]
Worked for me, thank you so much! Are you planning on keeping the override indefinitely or are you going to upgrade everything at some point

**Reply by madibaT** [reply]
I still need to check with our Frond-End developers. We will definitely need to upgrade at some point I think. Our application is still growing.

**Comment by zombieJ** [comment]
https://github.com/ant-design/ant-design-icons/issues/572#issuecomment-1560428091

**Reply by madibaT** [reply]
Thanks. I see the fix is merged now :)

**Comment by madibaT** [comment]
@zombieJ 's comment points to an open issue about this with a wider community. I will close this.
